### PR TITLE
feat(isort): Implement known-local-folder

### PR DESCRIPTION
**Pull request by spaceone**
sub item of #2419 

to make #2419 really happen some more refactoring has to be done by outsourcing the sections into mapping.

### CODE REVIEW

File: crates/ruff/src/rules/isort/categorize.rs [hunk 1]
```diff
@@ -43,16 +45,19 @@ pub fn categorize(
...
     known_first_party: &BTreeSet<String>,
     known_third_party: &BTreeSet<String>,
+    known_local_folder: &BTreeSet<String>,
     extra_standard_library: &BTreeSet<String>,
     target_version: PythonVersion,
 ) -> ImportType {
     let (import_type, reason) = {
         if level.map_or(false, |level| *level > 0) {
-            (ImportType::LocalFolder, Reason::NonZeroLevel)
+            (ImportType::Anything, Reason::NonZeroLevel)
```
Comment (spaceone) [inline crates/ruff/src/rules/isort/categorize.rs:54]: I didn't understand this one - a new name is necessary. Suggestions?

File: crates/ruff/src/rules/isort/categorize.rs [hunk 2]
```diff
@@ -43,16 +45,19 @@ pub fn categorize(
...
     let (import_type, reason) = {
         if level.map_or(false, |level| *level > 0) {
-            (ImportType::LocalFolder, Reason::NonZeroLevel)
+            (ImportType::Relative, Reason::NonZeroLevel)
         } else if known_first_party.contains(module_base) {
             (ImportType::FirstParty, Reason::KnownFirstParty)
         } else if known_third_party.contains(module_base) {
             (ImportType::ThirdParty, Reason::KnownThirdParty)
+        } else if known_local_folder.contains(module_base) {
+            (ImportType::LocalFolder, Reason::KnownLocalFolder)
```
Comment (charliermarsh) [inline crates/ruff/src/rules/isort/categorize.rs:60]: Doesn't this cause us to render the `.` imports and the "known local folder" imports as separate sections? Does that match isort's behavior? If so I find it a little surprising.

Comment (charliermarsh) [reply crates/ruff/src/rules/isort/categorize.rs:60]: Like, I would've expected that we remove `ImportType::Relative`, and just use `ImportType::LocalFolder` in both places.

Comment (spaceone) [reply crates/ruff/src/rules/isort/categorize.rs:59]: yes, done.

Comment (charliermarsh) [reply crates/ruff/src/rules/isort/categorize.rs:59]: Thx

**Pull request merged**